apm-sniffer/apm-sdk-plugin/mysql-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdbc/mysql/v6/ConnectionCreateOldInterceptor.java

```diff
@@ -41,7 +41,12 @@ public void beforeMethod(Class clazz, Method method, Object[] allArguments, Clas
     public Object afterMethod(Class clazz, Method method, Object[] allArguments, Class<?>[] parameterTypes,
         Object ret) {
         if (ret instanceof EnhancedInstance) {
-            ConnectionInfo connectionInfo = ConnectionCache.get(allArguments[1].toString(), allArguments[2].toString());
+            String database = "";
+            try {
+                Method getDbMethod = parameterTypes[0].getDeclaredMethod("getDatabase");
+                database = (String) getDbMethod.invoke(allArguments[0]);
+            } catch (Throwable t) { }
```

Comment:
Why do you need this reflector? You could covert `parameterTypes[0]` to the target type and call method directly

Reply:
Because this plugin already exists one mysql-connector-java dependency, if I add the old version dependency and use the target type directly, I won't be able to compile it. Do you have any suggestions?

Reply:
Do you mean this `getDatabase` method is not available in some versions?

Reply:
mysql-connector-java version 6.0.6 doesn't have parameterTypes[0] class (ie. ConnectionString), and version 6.0.2 doesn't have com.mysql.cj.core.conf.url.HostInfo class  (ConnectionCreateNewInterceptor needing it), but I need those two classes  to coexist if use the target type directly.

Reply:
If that method doesn't exist, how your fix is correct? You just try to catch a known exception, but don't help for the result.

Reply:
There are two Interceptor for two mysql-connector-java version in one plugin.My point is how to add two mysql-connector-java version dependencies for use the target type directly  and separately.

Reply:
They should be separated, and use witness class to identify the versions.

Reply:
However, Maven cannot add two versions  of mysql-connector-java dependencies. So I use the reflector.